ic-agent/src/agent/replica_api.rs

```diff
@@ -104,7 +104,7 @@ pub struct ReadStateResponse {
 
 /// A `Certificate` as defined in https://docs.dfinity.systems/public/#_certificate
 #[derive(Deserialize)]
-pub(crate) struct Certificate {
+pub struct Certificate {
```

Comment:
We don't want to expose anything from replica_api. Any structure in this file needs to stay private. That's because they are ADT, not proper user facing APIs. 

You should build an external facing Certificate which includes the methods you want.